llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2139,6 +2146,62 @@ class VPReductionRecipe : public VPSingleDefRecipe {
   }
 };
 
+/// A recipe to represent inloop reduction operations with vector-predication
+/// intrinsics, performing a reduction on a vector operand with the explicit
+/// vector length (EVL) into a scalar value, and adding the result to a chain.
+/// The Operands are {ChainOp, VecOp, EVL, [Condition]}.
+class VPReductionEVLRecipe : public VPSingleDefRecipe {
```

Comment:
I think we need to also add support for the new recipe to type inference in VPlanAnalysis.cpp

Reply:
f2b218320b0d1fcb9bfdeca8d3d8c6072e8057ab